x-pack/plugin/logsdb/src/main/java/org/elasticsearch/xpack/logsdb/SyntheticSourceLicenseService.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.logsdb;
+
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.license.License;
+import org.elasticsearch.license.LicensedFeature;
+import org.elasticsearch.license.XPackLicenseState;
+
+/**
+ * Determines based on license and fallback setting whether synthetic source usages should fallback to stored source.
+ */
+public final class SyntheticSourceLicenseService {
+
+    public static final String LOGSDB_FEATURE_FAMILY = "logsdb";
```

Comment:
As we discussed IMO this should be more generic too since it applies to TSDB as well. At this point i guess the entire plugin should be more like "synthetic source plugin"?

Reply:
> As we discussed IMO this should be more generic too since it applies to TSDB as well. 

👍 

> At this point i guess the entire plugin should be more like "synthetic source plugin"?

The idea is that this plugin will hold other logsdb features too.  With that in mind, I think the logsdb plugin name does make sense.